src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,524 @@
+                    {
+                        token = Token.Comma;
+                        return true;
+                    }
+                    case '=':
+                    {
+                        token = Token.Equals;
+                        return true;
+                    }
+                    case '\0':
+                    {
+                        token = Token.End;
+                        return true;
+                    }
+                }
+
+                if (!IsWhiteSpace(c))
+                {
+                    break;
+                }
+            }
+
+            using ValueStringBuilder sb = new ValueStringBuilder(stackalloc char[64]);
+
+            char quoteChar = '\0';
+            if (c == '\'' || c == '\"')
+            {
+                quoteChar = c;
+                if (!TryGetNextChar(out c))
+                {
+                    token = default;
+                    return false;
+                }
+            }
+
+            for (; ; )
+            {
+                if (c == 0)
+                {
+                    if (quoteChar != 0)
+                    {
+                        // EOS and unclosed quotes is an error
+                        token = default;
+                        return false;
+                    }
+                    // Reached end of input and therefore of string
+                    break;
+                }
+
+                if (quoteChar != 0 && c == quoteChar)
+                    break;  // Terminate: Found closing quote of quoted string.
+
+                if (quoteChar == 0 && (c == ',' || c == '='))
+                {
+                    _index--;
+                    break;  // Terminate: Found start of a new ',' or '=' token.
+                }
+
+                if (quoteChar == 0 && (c == '\'' || c == '\"'))
+                {
+                    token = default;
+                    return false;
+                }
+
+                if (c == '\\')
+                {
+                    if (!TryGetNextChar(out c))
+                    {
+                        token = default;
+                        return false;
+                    }
+
+                    switch (c)
+                    {
+                        case '\\':
+                        case ',':
+                        case '=':
+                        case '\'':
+                        case '"':
+                            sb.Append(c);
+                            break;
+                        case 't':
+                            sb.Append('\t');
+                            break;
+                        case 'r':
+                            sb.Append('\r');
+                            break;
+                        case 'n':
+                            sb.Append('\n');
+                            break;
+                        default:
+                            token = default;
+                            return false; //unreachable
```

Comment:
If this is actually unreachable, should we assert?

Reply:
It's my bad.  The code was previously using the comment "unreachable" to express that it was a line after a method that was always throwing. I've changed it to return false instead and it's absolutely reachable.

https://github.com/dotnet/runtime/blob/7d91bf56ec48329ac3ed65c35fdfa9b6fd81c2d3/src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs#L410-L412